server/consumer.go

```diff
@@ -786,15 +786,7 @@ func (mset *stream) addConsumerWithAssignment(config *ConsumerConfig, oname stri
 	o.nextMsgSubj = fmt.Sprintf(JSApiRequestNextT, mn, o.name)
 
 	// If the user has set the inactive threshold, set that up here.
-	if o.cfg.InactiveThreshold > 0 {
-		o.dthresh = o.cfg.InactiveThreshold
-	} else if !o.isDurable() {
-		// Ephemerals will always have inactive thresholds.
-		// Add in 1 sec of jitter above and beyond the default of 5s.
-		o.dthresh = JsDeleteWaitTimeDefault + time.Duration(rand.Int63n(1000))*time.Millisecond
-		// Only stamp config with default sans jitter.
-		o.cfg.InactiveThreshold = JsDeleteWaitTimeDefault
-	}
+	o.cfg.InactiveThreshold = o.updateInactiveThresholdLocked(o.cfg.InactiveThreshold, false)
```

Comment:
I have my doubts creating the timer from here. It looks to me that this function (or addConsumer) may be invoked from non consumer leader servers. Could it then lead a server from deleting the consumer because it thinks that there is no interest/pull requests? I need to run some tests...

Reply:
I did wonder about that, I assumed there was probably a reason for the unit test specifically testing that the timer wasn't already set but I don't really know enough about the cluster behaviour at this point.